sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -646,6 +651,15 @@ which is used in a graph will be kept alive on the host for the lifetime of the
 graph. Destroying that buffer during the lifetime of a `command_graph`
 constructed with this property results in undefined behavior.
 
+===== Disable-In-Order-Optimization Property [[disable-in-order-optimization]]
+
+The `property::graph::disable_in_order_optimization` property disables 
+an optimization which can be applied to in-order graph to 
+reduce its execution time.
+This property is passed to the `command_graph::finalize()` function.
+Note that this property must be passed to the finalize function if users 
+want to profile the exec_graph execution.
```

Comment:
This is quite a specific implementation detail to expose to the SYCL user, I think a better way to express the property would be `property::graph::enable_profiling` to align with queues, without going into the details of optimizations that will be disabled as there could be other optimizations in future that won't be compatible.

We should also specify an error is thrown when trying to profiling an event from a graph submission that was created without this property being set on the exec graph.

Reply:
I agree. A `property::graph::enable_profiling` could also cover other optimizations such as disabling profiling on events.